prov/sockets/src/sock_poll.c

```diff
@@ -142,8 +142,10 @@ static int sock_poll_poll(struct fid_poll *pollset, void **context, int count)
 						cntr_fid);
 			sock_cntr_progress(cntr);
 			pthread_mutex_lock(&cntr->mut);
-			if (atomic_get(&cntr->value) >=
-				atomic_get(&cntr->threshold)) {
+			if (atomic_get(&cntr->value) !=
+			    atomic_get(&cntr->last_read_val)) {
+				atomic_set(&cntr->last_read_val,
```

Comment:
Is this just trying to prevent the case where you don't need to write out the value if it is already set? 

In the case that they aren't equal, that's 3 calls to `atomic_get` and 1 to `atomic_set`. Since each of these is acquiring a lock it may just be quicker to _always_ just do: 

```c
atomic_set(&cntr->last_read_val, atomic_get(&cntr->value));
```

Of course that only works if it can always be updated.